src/pkg/services/m365/api/item_pager.go

```diff
@@ -111,18 +165,40 @@ func enumerateItems[T any](
 		// get the next page of data, check for standard errors
 		page, err := pager.GetPage(ctx)
 		if err != nil {
-			return nil, graph.Stack(ctx, err)
+			npr.err = graph.Stack(ctx, err)
+			return
 		}
 
-		result = append(result, page.GetValue()...)
+		npr.pages <- nextPage[T]{items: page.GetValue()}
```

Comment:
nit: just for safety could you please expand this (and other attempts to send on the pager channel that are in this file) to check for context cancellation ?
```suggestion
		select {
			case <-ctx.Done():
				// Some error handling/logging
				
			case npr.pages <- nextPage[T]{items: page.GetValue()}:
		}
```

Reply:
sure, can do